evalml/pipelines/time_series_pipeline_base.py

```diff
@@ -286,3 +306,25 @@ def _estimator_predict(self, features):
         This helper passes y as an argument if needed by the estimator.
         """
         return self.estimator.predict(features)
+
+    def dates_needed_for_prediction(self, date):
+        """Return dates needed to forecast the given date in the future.
+
+        Args:
+            date (np.datetime64,): Date to forecast in the future.
+
+        Returns:
+            dates_needed (tuple(np.datetime64)): Range of dates needed to forecast the given date.
+        """
+        beginning_date = date - np.timedelta64(
+            self.forecast_horizon
+            + self.max_delay  # include start delay for featurization
+            + self.gap  # add first gap for the actual gap from the end date
```

Comment:
We have a limitation s.t we require the training data (using the dates from this method) to be larger than the gap. Thus, I'm directly adding the gap here so that we ensure our training data is always long enough. This can be optimized further by the dates needed to push the length of the training data over the gap but decided to err on the side of caution for the first pass. I believe its safer to ask the user for more data unless we know that it is hard for the user to provide us such data.